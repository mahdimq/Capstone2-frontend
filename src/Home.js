import React, { useState } from 'react';

// Import Components
import Grid from './components/Grid';
import HeroImage from './components/HeroImage';
import LoadMore from './components/LoadMore';
import MovieTile from './components/MovieTile';
import Search from './components/Search';
import Spinner from './components/Spinner';

// Import config files to load home page
import {
	POPULAR_ENDPOINT,
	SEARCH_ENDPOINT,
	poster_size,
	backdrop_size,
	IMAGE_URL
} from './helpers/config';

// Import Custom Hook
import { useHomeHook } from './hooks/useHomeHook';

// show default image if poster image doesn't exist
import NoPoster from './images/no_poster.jpg';

const Home = () => {
	const [searchQuery, setSearchQuery] = useState('');
	const [
		{
			state: { movies, currentPage, totalPages, heroImage },
			loading,
			error
		},
		fetchMovies
	] = useHomeHook(searchQuery);

	// Search movies function
	const searchMovies = (query) => {
		const endpoint = query ? `${SEARCH_ENDPOINT}${query}` : POPULAR_ENDPOINT;

		setSearchQuery(query);
		fetchMovies(endpoint);
	};

	// Load more movies button
	const loadMoreMovies = () => {
		const searchEndpoint = `${SEARCH_ENDPOINT}${searchQuery}&page=${currentPage + 1}`;
		const popularEndpoint = `${POPULAR_ENDPOINT}&page=${currentPage + 1}`;

		const endpoint = searchQuery ? searchEndpoint : popularEndpoint;
		fetchMovies(endpoint);
	};

	if (movies.length === 0)
		return (
			<h1 style={{ color: '#c20a0a', margin: '1.2em auto', textAlign: 'center' }}>
				NO RESULTS FOUND <p>(Please refresh the page)</p>
				<Spinner />
			</h1>
		);
	if (!movies[0]) return <Spinner />;
	if (error) return <h3>Uh Oh! Something went wrong!</h3>;

	return (
		<>
			{!searchQuery && (
				<HeroImage
					image={`${IMAGE_URL}/${backdrop_size}/${heroImage.backdrop_path}`}
					title={heroImage.original_title}
					text={heroImage.overview}
				/>
			)}

			<Search callback={searchMovies} />

			<Grid header={searchQuery ? 'Search Results' : 'Popular Movies'}>
				{movies.map((film) => (
					<div key={film.id}>
						<MovieTile
							clickable={true}
							image={
								film.poster_path ? `${IMAGE_URL}/${poster_size}/${film.poster_path}` : NoPoster
							}
							movieId={film.id}
							movieTitle={film.original_title}
						/>
						<p className='movietile-title'>{film.title}</p>
					</div>
				))}
			</Grid>

			{loading && <Spinner />}

			{currentPage < totalPages && !loading ? (
				<LoadMore text='Load More' callback={loadMoreMovies} />
			) : (
				<h1 style={{ color: '#c20a0a', margin: '1.2em auto', textAlign: 'center' }}>
					END OF RESULTS
				</h1>
			)}
		</>
	);
};

export default Home;
