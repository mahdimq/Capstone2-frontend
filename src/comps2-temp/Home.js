// import React from 'react';
// import { Link } from 'react-router-dom';

// //Home page banner

// function Home() {
// 	return (
// 		<section>
// 			<h1>HOME COMPONENT</h1>

// 			<h1>Capstone</h1>
// 			<p>This is the home component</p>
// 			<Link to='/login'>Login</Link>
// 			<Link to='/signup'>Sign Up</Link>
// 		</section>
// 	);
// }

// export default Home;
