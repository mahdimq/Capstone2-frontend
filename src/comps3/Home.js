import React, { useState, useEffect } from 'react';
import { Link } from 'react-router-dom';
import MovieTile from '../components/MovieTile';

import CapstoneApi from '../CapstoneApi';

import Grid from '../components/Grid';

import NoPoster from '../images/no_poster.jpg';
//Home page banner
const IMAGE_URL = 'http://image.tmdb.org/t/p';

const poster_size = 'w500';

function Home() {
	const [comedy, setComedy] = useState({ movies: [] });

	const fetchComedy = async () => {
		try {
			const result = await CapstoneApi.getComedy();
			const randIdx = Math.floor(Math.random() * 20); //<-- to get random hero image from popular movies
			setComedy((prevMovies) => ({
				...prevMovies,
				movies: [...result.results],
				heroImage: prevMovies.heroImage || result.results[randIdx],
				currentPage: result.page,
				totalPages: result.total_pages
			}));
		} catch (err) {
			console.error(err);
		}
	};

	useEffect(() => {
		fetchComedy();
	}, []);

	return (
		<section>
			<h1>HOME COMPONENT</h1>
			<Grid header='Comedy Movies'>
				{comedy.movies.map((film) => (
					<MovieTile
						key={film.id}
						clickable={true}
						image={film.poster_path ? `${IMAGE_URL}/${poster_size}/${film.poster_path}` : NoPoster}
						movieId={film.id}
						movieTitle={film.original_title}
					/>
				))}
			</Grid>

			<Link to='/login'>Login</Link>
			<Link to='/signup'>Sign Up</Link>
		</section>
	);
}

export default Home;
