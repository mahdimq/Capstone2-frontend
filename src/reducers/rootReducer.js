import movieReducer from './movieReducer';
import userReducer from './userReducer';
import watchlistReducer from './watchlistReducer';
import alertsReducer from './alertsReducer';
import { combineReducers } from 'redux';

const rootReducer = combineReducers({
	user: userReducer,
	movies: movieReducer,
	watchlist: watchlistReducer,
	alerts: alertsReducer
});

export default rootReducer;
