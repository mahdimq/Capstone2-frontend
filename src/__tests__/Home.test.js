import React from 'react';
import { render } from '@testing-library/react';
import Home from '../Home';
import { Provider } from 'react-redux';
import rootReducer from '../reducers/rootReducer';
import { createStore } from 'redux';
import { BrowserRouter } from 'react-router-dom';

const store = createStore(rootReducer);

it('renders without crashing', function () {
	render(
		<Provider store={store}>
			<BrowserRouter>
				<Home />
			</BrowserRouter>
		</Provider>
	);
});

it('matches snapshot', function () {
	const { asFragment } = render(
		<Provider store={store}>
			<BrowserRouter>
				<Home />
			</BrowserRouter>
		</Provider>
	);
	expect(asFragment()).toMatchSnapshot();
});
