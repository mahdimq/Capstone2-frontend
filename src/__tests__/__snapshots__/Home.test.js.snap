// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <h1
    style="color: rgb(194, 10, 10); margin: 1.2em auto; text-align: center;"
  >
    NO RESULTS FOUND 
    <p>
      (Please refresh the page)
    </p>
    <div
      class="sc-iBPRYJ dgnvzH"
    />
  </h1>
</DocumentFragment>
`;
